ui/src/app/components/header/header.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from '@angular/core'
+import ISpannerConfig from '../../model/SpannerConfig'
+import { MatDialog } from '@angular/material/dialog'
+import { UpdateSpannerConfigFormComponent } from '../update-spanner-config-form/update-spanner-config-form.component'
+import { DataService } from 'src/app/services/data/data.service'
+import { SidenavService } from 'src/app/services/sidenav/sidenav.service'
+import { ClickEventService } from 'src/app/services/click-event/click-event.service'
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  spannerConfig: ISpannerConfig
+  isOfflineStatus: boolean = false
+  constructor(
+    private data: DataService,
+    private dialog: MatDialog,
+    private sidenav: SidenavService,
+    private clickEvent: ClickEventService
+  ) {
+    this.spannerConfig = { GCPProjectID: '', SpannerInstanceID: '' }
+  }
+
+  ngOnInit(): void {
+    this.data.config.subscribe((res: ISpannerConfig) => {
+      this.spannerConfig = res
+    })
+
+    this.data.isOffline.subscribe({
+      next: (res) => {
+        this.isOfflineStatus = res
+      },
+    })
+    this.clickEvent.spannerConfig.subscribe((res) => {
+      if (res) {
+        this.openEditForm()
+      }
+    })
+  }
+
+  openEditForm() {
+    let openDialog = this.dialog.open(UpdateSpannerConfigFormComponent, {
+      width: '30vw',
+      minWidth: '400px',
```

Comment:
using absolute sizes like this might lead to the UI being completely filled with this dialog. Is this as per the design ?

Reply:
Normally the width is set to 30% of the viewport. However the min width is also set so that the content (the form components) are not misplaced.

Reply:
Resolving this comment after today's discussion.